Assets/Scripts/UIController.cs

```diff
@@ -4,21 +4,134 @@
 
 public class UIController : MonoBehaviour
 {
-
     private const float FADE_ANIMATION_DURATION = 0.5f;
 
     private const float MESSAGE_DISPLAY_DURATION = 1.0f;
 
+    private IEnumerator messageCorutine;
+
     private Text messageText;
 
+    private Image[] lives;
```

Comment:
W Javie bym to zrobił listą. W C# czytelniej jednak wyglądają tablice, przynajmniej moim zdaniem  :P